### PR TITLE
Design product compare GraphQL schema

### PR DESCRIPTION
**Pull request by paliarush**
GraphQL schema for product compare

### CODE REVIEW

File: design-documents/graph-ql/coverage/product-compare.graphqls [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    addProductsToCompareList(compare_list_id: String!, skus: [String!]!): AddProductToCompareListOutput!
+    removeProductsFromCompareList(compare_list_id: String!, skus: [String!]!): RemoveProductsFromCompareListOutput!
+}
+
+type Query {
+    productCompareListId(): String! # ALways new ID for guests. A new or an existing ID for customers
+    productCompareList(compare_list_id: String!): ProductCompareList!
+}
+
+input ProductCompareList {
```
Comment (akaplya) [inline design-documents/graph-ql/coverage/product-compare.graphqls]: should be `type`?

File: design-documents/graph-ql/coverage/product-compare.graphqls [hunk 2]
```diff
@@ -0,0 +1,22 @@
+type Mutation {
+    addProductsToCompareList(compare_list_id: String!, skus: [String!]!): AddProductToCompareListOutput!
+    removeProductsFromCompareList(compare_list_id: String!, skus: [String!]!): RemoveProductsFromCompareListOutput!
+}
+
+type Query {
+    productCompareListId(): String! # ALways new ID for guests. A new or an existing ID for customers
```
Comment (akaplya) [inline design-documents/graph-ql/coverage/product-compare.graphqls]: can we use `ID` instead

File: design-documents/graph-ql/coverage/product-compare.graphqls (hunk 1 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/product-compare.graphqls]: yes

File: design-documents/graph-ql/coverage/product-compare.graphqls (hunk 2 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/product-compare.graphqls]: Okay, makes sense.

**Pull request merged**